// // put(key, value)
// bool put(const std::string &key, const std::string &value);
// // value = get(key)
// std::string get(const std::string &key);
// // del(key)
// bool del(const std::string &key);

// // clear memtable and disk
// void reset();
// // flush memtable to disk
// void flush();
syntax = "proto3";
package kv;
option cc_generic_services = true;

service KVService {
  rpc Put(PutRequest) returns (PutResponse) {}
  rpc Get(GetRequest) returns (GetResponse) {}
  rpc Del(DelRequest) returns (DelResponse) {}
  rpc Reset(ResetRequest) returns (ResetResponse) {}
  rpc Flush(FlushRequest) returns (FlushResponse) {}
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  bool ok = 1;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
}

message DelRequest {
  string key = 1;
}

message DelResponse {
  bool ok = 1;
}

message ResetRequest {}

message ResetResponse {
  bool ok = 1;
}

message FlushRequest {}

message FlushResponse {
  bool ok = 1;
}